### PR TITLE
vpn, dappctrl dockerfiles and docker-compose file

### PR DESCRIPTION
**Pull request by furkhat**
Images are hosted at https://hub.docker.com/u/privatix/dashboard/

### CODE REVIEW

File: vpn.dockerfile [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+RUN cp ./pki/issued/myvpn.crt /etc/openvpn/config/
+RUN cp ./pki/private/myvpn.key /etc/openvpn/config/
+
+
+WORKDIR /etc/openvpn/config
+RUN openssl dhparam -out dh2048.pem 2048
+RUN openvpn --genkey --secret ta.key
+
+## create config
+RUN echo "port 1194" >> server.conf
```
Comment (ababo) [inline vpn.dockerfile]: Did you consider multiline input using `cat`?

File: README.md [hunk 2]
```diff
@@ -59,6 +59,23 @@ Build OpenVPN session trigger:
 go install $DAPPCTRL/tool/dapptrig
 ```
 
+## Using docker
```
Comment (drew2a) [inline README.md]: 👍

File: vpn.dockerfile (hunk 1 above)
Comment (ababo) [reply vpn.dockerfile]:
```bash
cat > FILE.txt <<EOF

info code info 
info code info
info code info

EOF 
```

File: README.md [hunk 3]
```diff
@@ -59,6 +59,23 @@ Build OpenVPN session trigger:
...
+We have prepared two images and a compose file to make it easier to run app and its dependencies.
+There are 3 services in compose file:
+  1. db - uses public postgres image
```
Comment (drew2a) [inline README.md]: Can we use markdown and em-dash for cases like this?
Example: `db` — uses public postgres image

File: README.md [hunk 4]
```diff
@@ -59,6 +59,23 @@ Build OpenVPN session trigger:
...
+  2. vpn - image `privatix/dapp-vpn-server` is an openvpn that uses `dapptrig` for auth, connect and disconnect
+  3. dappctrl - image `privatix/dappctrl` is a main controller app
+
+If you want to develop dappctrl then it is convenient to run its dependencies using docker, but controller itself at your host machine:
+```
+docker-compose up vpn db
+```
+
```
Comment (drew2a) [inline README.md]: redundant \n
:)

File: vpn.dockerfile (hunk 1 above)
Comment (furkhat) [reply vpn.dockerfile]: I can do:
```
RUN bash -c "$(/bin/echo -e "cat > server.conf <<EOM\
\n line\
\n another line\
\n##################
\nEOM\n")
```
Hm, I don't know. It's kinda noisy. Do you like this better?

Comment (ababo) [reply vpn.dockerfile]: Take a look at the code I've googled:
```bash
RUN <<EOR
    cd /
    git clone https://github.com/
    cd /fileadmin/
    bundle install
    rake db:migrate
    bundle exec rails runner "eval(File.read 'createTestUser.rb')"
    mkdir /pending
    mkdir /live
    chmod 777 /pending
    chmod 777 /live
EOR
```

Comment (furkhat) [reply vpn.dockerfile]: hm, that gives a syntax error.

Comment (ababo) [reply vpn.dockerfile]: Yes, now I see that was just a proposal. Here is the working example: https://stackoverflow.com/questions/33439230/how-to-write-commands-with-multiple-lines-in-dockerfile-while-preserving-the-new

Comment (ababo) [reply vpn.dockerfile]: Doesn't seem to be enough elegant but can drastically reduce file size anyway.

Comment (furkhat) [reply vpn.dockerfile]: updated.

**Pull request merged**